src/select/Select.ts

```diff
@@ -226,8 +227,15 @@ export default class Select extends SelectBase<SelectProperties> {
 		} = this.properties;
 
 		// update custom option registry
-		if ( !this.registry || includes(evt.changedPropertyKeys, 'customOption')) {
-			this.registry = this._createRegistry(customOption);
+		if ( !this._registry || includes(evt.changedPropertyKeys, 'customOption')) {
+			const registry = this._createRegistry(customOption);
+			if (this._registry) {
+				this.registries.replace(this._registry, registry);
```

Comment:
What's the purpose of `registries` now? I couldn't find anything in the README of widget-core yet.